src/game/SpellEffects.cpp

```diff
@@ -2572,14 +2556,20 @@ void Spell::EffectSummonGuardian(SpellEffectIndex eff_idx)
         if (m_caster->FindGuardianWithEntry(pet_entry))
             return;                                         // find old guardian, ignore summon
 
-    // in another case summon new
-    uint32 level = m_caster->getLevel();
+    // Level of pet summoned
+    uint32 level = std::max(m_caster->getLevel() + m_spellInfo->EffectMultipleValue[eff_idx], 1.0f);
```

Comment:
Point taken, this very line is not 'my code' per say, though; https://github.com/cmangos/mangos-classic/commit/a30ffd97790515b2a2dc7f1558df77f4437ea048